engine/execution/ingestion/stop_control.go

```diff
@@ -0,0 +1,235 @@
+package ingestion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// StopControl is a specialized component used by ingestion.Engine to encapsulate
+// control of pausing/stopping blocks execution.
+// It is intended to work tightly with the Engine, not as a general mechanism or interface.
+// StopControl follows states described in StopState
+type StopControl struct {
+	sync.RWMutex
+	// desired stop height, the first value new version should be used, so this height WON'T
```

Comment:
Reading this highlighted "WON'T", I wonder if it makes sense to add some sanity checks in `BlockExecuted` and `executingBlockHeight` and `blockExecuted` event to if stop height is set then we never execute blocks above or equal to that height?

Reply:
I'm not sure what we would gain from this.
Its too late to halt execution in those methods, and the one which allow skipping blocks already has this check.